### DISCUSSION TITLE
NX partitionning problem with Kirkstone build - UDA

### DISCUSSION THREAD

**Discussion by andregirard37** [OE4T/meta-tegra#1491] [Q&A]
Hello,

I am migrating my Yocto build from Dunfell to Kirkstone. We have applications running on a custom partition, let's call it APPS. For that, we are using the UDA partition to give us the remaining space on the device, as described by:

        <partition name="UDA" type="data">
            <allocation_policy> sequential </allocation_policy>
            <filesystem_type> basic </filesystem_type>
            <size> 18432 </size>
            <file_system_attribute> 0 </file_system_attribute>
            <allocation_attribute> 0x808 </allocation_attribute>
            <percent_reserved> 0 </percent_reserved>
            <description> **Required.** **Automatically takes all remaining space on the device** except that
              occupied by the `secondary_gpt` partition. Allocation attribute must be set to 0x808.
              May be mounted and used to store user data. `/dev/mmcblk0p22`</description>
        </partition>

On the Dunfell build, it would give us 5Gbytes or so of space. But for some reasons on the Kirkstone build, we only get 101MBytes left:

lsblk
NAME         MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
mmcblk0      179:0    0 14.7G  0 disk
|-mmcblk0p1  179:1    0    8G  0 part /
|-mmcblk0p2  179:2    0   64M  0 part
|-mmcblk0p3  179:3    0  448K  0 part
|-mmcblk0p4  179:4    0 32.5M  0 part
|-mmcblk0p5  179:5    0  2.5M  0 part
|-mmcblk0p6  179:6    0   64K  0 part
|-mmcblk0p7  179:7    0    1M  0 part
|-mmcblk0p8  179:8    0    1M  0 part
|-mmcblk0p9  179:9    0    1M  0 part
|-mmcblk0p10 179:10   0  1.5M  0 part
|-mmcblk0p11 179:11   0    1M  0 part
|-mmcblk0p12 179:12   0   64M  0 part
|-mmcblk0p13 179:13   0  448K  0 part
|-mmcblk0p14 179:14   0 32.5M  0 part
|-mmcblk0p15 179:15   0   80M  0 part
|-mmcblk0p16 179:16   0  512K  0 part
|-mmcblk0p17 179:17   0  100M  0 part
|-mmcblk0p18 179:18   0   64M  0 part /opt/nvidia/esp
|                                                             /boot/efi
|-mmcblk0p19 179:19   0   80M  0 part
|-mmcblk0p20 179:20   0  512K  0 part
|-mmcblk0p21 179:21   0   64M  0 part
|-mmcblk0p22 179:22   0   101M  0 part /apps

Adding up all the SIZE above, we are far from the 14.7G, so I don't understand why mmcblk0p22 is not way greater than 101M. Is there some reserved space I am aware of or is there something I am missing on the flash.xml.in ?

Thank you!
Andre

**Comment by madisongh** [comment]
Are there any entries in the storage layout XML that have a `<start_location>` element?  Some of the stock L4T layouts include that to reserve a chunk of empty space for future expansion of boot-related partitions.

**Reply by andregirard37** [reply]
I don't see any. However, after searching more online, it appears I can just create a new partition elsewhere in the flash.xml.in. I created a 1GByte partition above the "reserved_for_chain_A_user" partition and it worked. For some reason if I try to create a partition right before the UDA partition, then I am very limited in size.

**Reply by madisongh** [reply]
If you've got `parted` or `sgdisk` (from the `gptfdisk` package) in your image, you could use one of those tools to examine the partition offsets and sizes
